tests/app/views/test_suppliers.py

```diff
@@ -118,9 +106,124 @@ def test_query_string_prefix_page_out_of_range(self):
 
         assert_equal(response.status_code, 400)
-        assert_in(b'Invalid page argument', response.get_data())
+
+
+class TestPutSupplier(BaseApplicationTest, JSONUpdateTestMixin):
+    method = "put"
+    endpoint = "/suppliers/123456"
+
+    def setup(self):
+        super(TestPutSupplier, self).setup()
+
+    def put_import_supplier(self, supplier):
+        return self.client.put(
+            '/suppliers/' + str(supplier.get('id', 0)),
+            data=json.dumps({
+                'suppliers': supplier
+            }),
+            content_type='application/json')
+
+    def test_add_a_new_supplier(self):
+        with self.app.app_context():
+            payload = self.load_example_listing("Supplier")
+            response = self.put_import_supplier(payload)
+            assert_equal(response.status_code, 201)
+            supplier = Supplier.query.filter(
+                Supplier.supplier_id == 123456
+            ).first()
+            assert_equal(supplier.name, payload['name'])
+
+    def test_cannot_put_to_root_suppliers_url(self):
+        payload = self.load_example_listing("Supplier")
+        response = self.client.put(
+            '/suppliers',
+            data=json.dumps({
+                'suppliers': payload
+            }),
+            content_type='application/json')
+
+        assert_equal(response.status_code, 405)
+
+    def test_supplier_json_id_does_not_match_route_id_parameter(self):
+        payload = self.load_example_listing("Supplier")
+        response = self.client.put(
+            '/suppliers/12345678901234567890',
+            data=json.dumps({
+                'suppliers': payload
+            }),
+            content_type='application/json')
+
+        assert_equal(response.status_code, 400)
+
+    def test_when_supplier_has_missing_contact_information(self):
+        payload = self.load_example_listing("Supplier")
+        payload.pop('contactInformation')
+        response = self.put_import_supplier(payload)
+
+        assert_equal(response.status_code, 400)
+
+    def test_when_supplier_has_missing_keys(self):
+        payload = self.load_example_listing("Supplier")
+        payload.pop('id')
+        payload.pop('name')
+        response = self.put_import_supplier(payload)
+        assert_equal(response.status_code, 400)
+
+    def test_when_supplier_contact_information_has_missing_keys(self):
+        payload = self.load_example_listing("Supplier")
+
+        payload['contactInformation'][0].pop('email')
+        payload['contactInformation'][0].pop('postcode')
+        payload['contactInformation'][0].pop('contactName')
+
+        response = self.put_import_supplier(payload)
+        assert_equal(response.status_code, 400)
+
+    def test_when_supplier_has_extra_keys(self):
+        payload = self.load_example_listing("Supplier")
+
+        payload.update({'newKey': 1})
+
+        response = self.put_import_supplier(payload)
+        assert_equal(response.status_code, 400)
```

Comment:
There are lots of ways to get a 400 response from the API.  If each of these different errors returns a specific error message at the point when the 400 is triggered then the tests should check that the error message is the one expected.  Otherwise the tests could continue to pass (i.e. the API returns 400) but for a different reason than the one they are supposed to be testing.